source/StartConditionsPanel.cpp

```diff
@@ -0,0 +1,292 @@
+	
+	GameData::Background().Draw(Point(), Point());
+	GameData::Interfaces().Get("menu background")->Draw(info, this);
+	GameData::Interfaces().Get("start conditions menu")->Draw(info, this);
+	GameData::Interfaces().Get("menu start info")->Draw(info, this);
+	
+	// TODO: Prevent text from overflowing.
+	descriptionWrappedText.Draw(
+		Point(descriptionBox.Left(), descriptionBox.Top() + descriptionScroll),
+		bright
+	);
+	
+	Point point(entryListBox.Left(), entryListBox.Top() - listScroll);
+	const Point offset = entryInternalBox.Dimensions() * .5;
+	
+	const Font &font = FontSet::Get(14);
+	for(const auto &it : GameData::StartOptions())
+	{
+		Rectangle zone(
+			point + offset, 
+			entryBox.Dimensions()
+		);
+		if(point.Y() > entryListBox.Bottom() || point.Y() < entryListBox.Top())
+		{
+			// Don't bother drawing if the item is above or under the list.
+			point += Point(0., entryBox.Height());
+			continue;
+		}
+		
+		bool isHighlighted = (&it == chosenStart);
+		
+		if(isHighlighted)
+			FillShader::Fill(zone.Center(), zone.Dimensions(), selectedBackground);
+		
+		DisplayText name = DisplayText(it.GetName(), Truncate::BACK);
+		font.Draw(name, point, isHighlighted ? bright : medium);
+		point += Point(0., entryBox.Height());
+	}
+}
+
+
+
+bool StartConditionsPanel::Drag(double dx, double dy)
+{
+	if(entryListBox.Contains(hoverPoint))
+	{
+		listScroll -= dy;
+		listScroll = min(entryBox.Height() * (GameData::StartOptions().size()-1), listScroll); 
+		listScroll = max(0., listScroll);
+	}
+	// TODO: When #4123 gets merged, re-add scrolling support.
+	// Right now it's pointless because it would make the text overflow.
+	
+	return true;
+}
+
+
+
+bool StartConditionsPanel::Scroll(double dx, double dy)
+{
+	return Drag(0., dy * Preferences::ScrollSpeed());
+}
+
+
+
+bool StartConditionsPanel::Hover(int x, int y)
+{
+	hoverPoint = Point(x, y);
+	return true;
+}
+
+
+
+bool StartConditionsPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, bool isNewPress)
+{
+	
+	if(key == 'b' || key == SDLK_ESCAPE || command.Has(Command::MENU) || (key == 'w' && (mod & (KMOD_CTRL | KMOD_GUI))))
+		GetUI()->Pop(this);
+	else if(key == SDLK_UP || key == SDLK_DOWN || key == SDLK_PAGEUP || key == SDLK_PAGEDOWN) 
+	{
+		int offset = 0;
+		offset += (key == SDLK_DOWN) - (key == SDLK_UP);
+		offset += PAGE_INTERVAL * ((key == SDLK_PAGEDOWN) - (key == SDLK_PAGEUP));
+		
+		listScroll += offset * entryBox.Height();
+		listScroll = min(max(listScroll, 0.), (GameData::StartOptions().size() - 1) * entryBox.Height());
+		
+		if(GameData::StartOptions().begin() - chosenStartIterator < offset)
+			chosenStartIterator = GameData::StartOptions().begin();
+		else if(GameData::StartOptions().end() - chosenStartIterator < offset)
+			chosenStartIterator = GameData::StartOptions().end() - 1;
+		else
+			chosenStartIterator += offset;
+		
+		chosenStart = &*chosenStartIterator;
+		descriptionWrappedText.Wrap(chosenStart->GetDescription());
+		
+		return false;
```

Comment:
Returning `false` from `Panel::KeyDown` (and other event handlers) means that the UI should pass that event to the next Panel in the stack, if the current panel doesn't trap all events. Please re-evaluate all the added event handler methods with this behavior of true/false return in mind.